magefiles/rulesengine/repos/infra_deployments.go

```diff
@@ -107,39 +106,32 @@ var InfraDeploymentsMultiPlatformComponentChangeRule = rulesengine.Rule{Name: "I
 	})}}
 
 var InfraDeploymentsBuildTemplatesComponentChangeRule = rulesengine.Rule{Name: "Infra-deployments PR Build-templates component File Change Rule",
-	Description: "Map build-templates tests files when Build-templates component files are changed in the infra-deployments PR",
+	Description: "Map build-templates tests files when build-pipeline-config.yaml is changed",
 	Condition: rulesengine.ConditionFunc(func(rctx *rulesengine.RuleCtx) bool {
-
-		return len(rctx.DiffFiles.FilterByDirGlob("components/build-service/base/build-pipeline-config/*")) != 0
-	}),
-	Actions: []rulesengine.Action{rulesengine.ActionFunc(func(rctx *rulesengine.RuleCtx) error {
-		AddLabelToLabelFilter(rctx, "build-templates")
-		return nil
-	})}}
-
-var InfraDeploymentsBuildServiceComponentChangeRule = rulesengine.Rule{Name: "Infra-deployments PR Build service component File Change Rule",
-	Description: "Map build service tests files when Build service component files are changed in the infra-deployments PR",
-	Condition: rulesengine.ConditionFunc(func(rctx *rulesengine.RuleCtx) bool {
-
-		return len(rctx.DiffFiles.FilterByDirGlob("components/build-service/**/*")) != 0 &&
-			len(rctx.DiffFiles.FilterByDirGlob("components/build-service/base/build-pipeline-config/*")) == 0
+		return len(rctx.DiffFiles.FilterByDirGlob("components/build-service/base/build-pipeline-config/build-pipeline-config.yaml")) != 0
 	}),
-	Actions: []rulesengine.Action{rulesengine.ActionFunc(func(rctx *rulesengine.RuleCtx) error {
-		AddLabelToLabelFilter(rctx, "build-service")
-		return nil
-	})}}
+	Actions: []rulesengine.Action{
+		rulesengine.ActionFunc(func(rctx *rulesengine.RuleCtx) error {
+			AddLabelToLabelFilter(rctx, "build-templates")
+			return nil
+		}),
+	},
+}
 
-var InfraDeploymentsBuildServiceTemplatesComponentChangeRule = rulesengine.Rule{Name: "Infra-deployments PR Build service component and templates File Change Rule",
-	Description: "Map build service tests files when Build service component and templates files are changed in the infra-deployments PR",
+var InfraDeploymentsBuildServiceComponentChangeRule = rulesengine.Rule{Name: "Infra-deployments PR Build Service component File Change Rule (excluding build-pipeline-config)",
+	Description: "Map build service tests files when files in build-service are changed except build-pipeline-config.yaml",
 	Condition: rulesengine.ConditionFunc(func(rctx *rulesengine.RuleCtx) bool {
 
-		return len(rctx.DiffFiles.FilterByDirGlob("components/build-service/**/*")) != 0 &&
-			len(rctx.DiffFiles.FilterByDirGlob("components/build-service/base/build-pipeline-config/*")) != 0
+		return len(rctx.DiffFiles.FilterByDirGlob("components/build-service/**/*")) > len(rctx.DiffFiles.FilterByDirGlob("components/build-service/base/build-pipeline-config/*")) &&
+			len(rctx.DiffFiles.FilterByDirGlob("components/build-service/**/*")) > 0
```

Comment:
Would it still work without this: `&& len(rctx.DiffFiles.FilterByDirGlob("components/build-service/**/*")) > 0`?

Reply:
@psturc , 
I ran the same scenarios without , it works too 
`&& len(rctx.DiffFiles.FilterByDirGlob("components/build-service/**/*")) > 0`
I'll remove and push here, wdyt?

Reply:
yep, I'd say go for it